### PR TITLE
validate names, conservatively: ascii, 32 char

### LINKED ISSUE #10: validate names helper
almost all of our `name` arguments have the same general format:

```perl
/^[a-z0-9][a-z0-9-_\.]*/i
```

given that many of our names are used as part of a path, we can further restrict the `*`

```perl
/^[a-z0-9][a-z0-9-_\.]{0,254}/i
```

to FreeBSD's `NAME_MAX` (255).

****
> edit with conclusions from the comments

HOWEVER, until FreeBSD 12, `fstab`'s mount paths are further restricted to 88, and not 255. Since we're supporting freebsd all the way down to 9.3, we should also limit our names.

As we've seen from experience from using UUIDs as names, 32 is a length that can stretch it. I suggest only using half of that (16)

```perl
/^[a-z0-9][a-z0-9-_\.]{0,15}/i
```

### PR DESCRIPTION
**Pull request by igalic**
This fixes #10.

### CODE REVIEW

File: libiocage/lib/Datasets.py [hunk 1]
```diff
@@ -135,6 +135,9 @@ def _get_or_create_dataset(self,
                                pool=None,
                                mountpoint=None):
 
+        if not libiocage.lib.helpers.validate_name(name):
+            raise NameError(f"Unable to validate 'name' ({name}) for Dataset.")
```
Comment (igalic) [inline libiocage/lib/Datasets.py]: these should be consistent with the other message

File: libiocage/lib/helpers.py [hunk 2]
```diff
@@ -95,6 +96,13 @@ def _prettify_output(output):
     ))
 
 
+# helper function to validate names
+def validate_name(name):
+    validate = re.compile(r'^[a-z0-9][a-z0-9\.-_]{0,31}$', re.I)
+    validate.fullmatch(name)
+    return True if validate.fullmatch(name) else False
```
Comment (gronke) [inline libiocage/lib/helpers.py]: Can be reduced to `bool(validate.fullmatch(name))`

File: libiocage/lib/helpers.py [hunk 3]
```diff
@@ -95,6 +96,13 @@ def _prettify_output(output):
     ))
 
 
+# helper function to validate names
+def validate_name(name):
+    validate = re.compile(r'^[a-z0-9][a-z0-9\.-_]{0,31}$', re.I)
```
Comment (gronke) [inline libiocage/lib/helpers.py]: You do a .fullmatch later on, so you can omit `^` and `$`

**Comment by igalic** [comment]
thanks for your review, @gronke.
I'll fix this up tomorrow on the train

File: libiocage/lib/Release.py [hunk 4]
```diff
@@ -36,6 +36,9 @@ def __init__(self, name=None,
         libiocage.lib.helpers.init_zfs(self, zfs)
         libiocage.lib.helpers.init_host(self, host)
 
+        if not libiocage.lib.helpers.validate_name(name):
+            raise NameError(f"Invalid 'name' for Release: '{name}'")
```
Comment (skarekrow) [inline libiocage/lib/Release.py]: Why are we validating RELEASE? They aren't created by a user with a name, they come from the project. We should remove this.

Comment (gronke) [inline libiocage/lib/Release.py]: Releases are stored with the same mechanism as Jails are, so that the same limitations apply.

**Comment by gronke** [comment]
@igalic ready to merge! Can you please rebase?

File: libiocage/lib/Release.py (hunk 4 above)
Comment (skarekrow) [inline libiocage/lib/Release.py]: I wasn't thinking right, @gronke is correct. Free to merge @igalic

Comment (igalic) [inline libiocage/lib/Release.py]: validate all the things, imo

**Pull request merged**